vit_unetr_ssl/transforms.py

```diff
@@ -0,0 +1,224 @@
+    """
+    Define MONAI Transforms for Training/Validation of the self-supervised pretrained model
+    :args: keys: list of keys to be used for the transforms, e.g. ["image", "label"]
+    :args: spatial_size: spatial size of the input image, e.g. (64, 256, 256)
+    :args: roi_size: spatial size of the region of interest, e.g. (64, 64, 64)
+    :args: number_of_holes: number of holes to be used for the RandCoarseDropoutd and RandCoarseShuffled transforms
+    """
+    transforms = Compose(
+        [
+            # Load image data using the key "image"
+            LoadImaged(keys=keys, image_only=True),
+            # Ensure that the channel dimension is the first dimension of the image tensor.
+            EnsureChannelFirstd(keys=keys),
+            # Ensure that the image orientation is consistently RPI
+            Orientationd(keys=keys, axcodes="RPI"),
+            # Resample the images to a specified pixel spacing
+            # NOTE: spine interpolation with order=2 is spline, order=1 is linear
+            Spacingd(keys=keys, pixdim=(1.0, 1.0, 1.0), mode=(2, 1)),
+            # Normalize the intensity values of the image
+            NormalizeIntensityd(keys=["image"], nonzero=False, channel_wise=False),
+            # Remove background pixels to focus on regions of interest.
+            # CropForegroundd(keys=["image"], source_key="image"),
+            # Pad the image to a specified spatial size if its size is smaller than the specified dimensions
+            ResizeWithPadOrCropd(keys=keys, spatial_size=spatial_size),
+            AsDiscreted(keys='label', to_onehot=None, threshold=0.5),
+            # Randomly crop samples of a specified size around the label (spinal cord)
+            # Note that it seems that the transform randomly selects a foreground point from image, then use it as
+            # center crop. This means that it can find the closest voxel that is just outside the SC and use it as the
+            # center (hence it includes the SC)
+            # https://github.com/Project-MONAI/MONAI/issues/452#issuecomment-636065502
+            RandCropByPosNegLabeld(
+                keys=keys,
+                label_key="label",
+                spatial_size=roi_size,
+                pos=2,
+                neg=1,
+                num_samples=2,
+                image_key="image",
+                image_threshold=0,
+            ),
+            # Create copies of items in the dictionary under new keys, allowing for the same image to be manipulated
+            # differently in subsequent transformations
+            CopyItemsd(keys=["image"], times=2, names=["gt_image", "image_2"], allow_missing_keys=False),
+
+            # AUGMENTED VIEW 1
+            OneOf(
+                transforms=[
+                    # Randomly drop regions of the image
+                    # 'dropout_holes=True': the dropped regions will be set to zero, introducing regions of no
+                    # information within the image.
```

Comment:
so I think this definition is incorrect -- if `dropout_holes=True` then the `holes` will be dropped out and the regions in the image with dropped holes will be filled with `fill_value` (i.e. they are not always set to zero, depends on `fill_value`)

Reply:
Good catch! You are right!

From [RandCoarseDropoutd](https://docs.monai.io/en/latest/transforms.html#randcoarsedropoutd) documentation:

> dropout_holes – if True, dropout the regions of holes and fill value, if False, keep the holes and dropout the outside and fill value. default to True.

Since, I do not specify `fill_value`, `None` is used, meaning:

> if None, will compute the min and max value of input image then randomly select value to fill, default to None.

I further debugged and plotted the transforms. It confirms your findings! 

<img width="626" alt="image" src="https://github.com/ivadomed/model-seg-dcm/assets/39456460/20346e03-e046-4bee-8588-a7389117735b">

Reply:
So regarding this -- I think it's best to set `fill_value=0`. Reason -- when the dropped out holes are randomly filled with values, then this makes it difficult for the model to learn (i.e. it is trying to learn that "there is a patch of random values, I have to figure out how to understand what's behind them"). 
INSTEAD, if we set `fill_value=0` then we're making the task simpler -- the models sees only black regions and it knows that these patches are uniform (i.e. filled only with zeros) instead of seeing that there are `n` holes and each hole has different values of pixels)

Bottomline -- set `fill_value=0` to make the training objective easier

Reply:
Thanks for this suggestion! Makes sense! Trying `fill_value=0` now (https://github.com/ivadomed/model-seg-dcm/pull/10/commits/bdb2ac155cb53b64423d7b7cb8e416492629707e).

<img width="649" alt="image" src="https://github.com/ivadomed/model-seg-dcm/assets/39456460/11ba6eb3-7053-4511-bf0b-0f21dd481a65">

Reply:
Thanks for the figures! maybe you should keep both `dropout_holes=False` and `dropout_holes=True` but with `fill_value=0`.

Reply:
Good point! I'm now showing both `dropout_holes=False` and `dropout_holes=True` for both `fill_value=None` and `fill_value=0`:

<img width="632" alt="image" src="https://github.com/ivadomed/model-seg-dcm/assets/39456460/778081ae-d20d-43ed-b643-e2ce3ed638ff">

Reply:
so we can go ahead with the `fill_value=0` set of transforms? what do you think?

Reply:
Yes! I pushed `fill_value=0` in https://github.com/ivadomed/model-seg-dcm/pull/10/commits/bdb2ac155cb53b64423d7b7cb8e416492629707e. The model is currently training!